pkg/common/release.go

```diff
@@ -60,3 +65,23 @@ func GetLatestReleaseInfo() (string, ReleaseMetadata) {
 
 	return latestRelease, ReleaseInfo[latestRelease]
 }
+
+func GetOrWriteReleaseInfo(path string) (string, error) {
+	filePath := filepath.Join(filepath.Clean(path), "release.json")
+	if _, err := os.Stat(filePath); err == nil {
+		data, err := os.ReadFile(filepath.Clean(filePath))
+		if err != nil {
+			return "", err
+		}
+		if err := unmarshal(data); err != nil {
+			return "", err
+		}
+		return fmt.Sprintf("Release file found at %s", filePath), nil
+	}
+
+	if err := os.MkdirAll(filepath.Clean(path), os.ModeDir|os.ModePerm); err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("Release file written to %s", filePath), os.WriteFile(filePath, releaseInfoFile, 0o644) // #nosec G306
```

Comment:
Did not understand this line. If we are able to find the release file, why write it to a new path ?
Is it for having the latest release written locally and access it later ?